apps/admin-server/src/hooks/use-status.tsx

```diff
@@ -0,0 +1,21 @@
+import useSWR from 'swr';
+
+export default function useStatuses(projectId?: string, id?: string) {
+  const url = `/api/openstad/api/project/${projectId}/status/${id}`;
+
+  const statuseswr = useSWR(projectId && id ? url : null);
+
+  async function updateStatus(name: string, seqnr: number, backgroundColor: string | undefined, color: string | undefined, label: string | undefined, mapIcon: string | undefined, listIcon: string | undefined, extraFunctionality: { editableByUser: boolean | undefined, noComment: boolean | undefined } ) {
```

Comment:
Zijn bij het updaten naam, seqnr en extraFunctionality niet ook optioneel? en voor een verkorting van de types die undefined kunnen zijn, zou je dan niet ?:string, ?:boolean enz gebruiken, maakt het wat overzichtelijker naar mijn idee? Bij tags kon extrafunctionality ook null zijn

Reply:
Ja, naam en seqnr moeten inderdaad ook optioneeel zijn. Gefixed (en dan ook gelijk in tags :)

Maar ?:xxx klopt niet: dat is fundamenteel iets anders. Je maakt dan het meesturen van de paramater optioneel, en omdat het geen named params in een object zijn maar een lijst van prams op volgorde, gaat dat hele rare effecten geven. Als je een dan een parameter weg laat dan wordt de volgende paramater de waarde die je daar instopt:
```
updateStatus(name?: string, seqnr?: number, backgroundColor?: string, color?: string)
```
kan ik aanroepen met
```
updateStatus('Naam', 10, 'de-kleur-waarde')
```
Die stopt dan de waarde die ik voor kleur wilde gebruiken in de backrgoundColor. Het is juist dat soort ellende die je met typing wilt voorkomen, lijkt me.

Reply:
Goed punt